### ISSUE TITLE
Android: open app UI when user answer the call, instead of the native dialer phone

### ISSUE THREAD

**Issue by namnm** [react-native-webrtc/react-native-callkeep#209]
Hi. Is it possible to open the app on Android when user press on answer call, like what we currently have on iOS?
Right now on Android, when user answer the call, the native dialer phone shows up. But for example, Skype/Messenger can show their app UI instead.

Thanks guys for the effort on this package.

**Comment by sboily** [comment]
Hello, callkeep is a wrapper for connectionService and callkit so it's not intent to do what you want. You need to check incallService https://developer.android.com/reference/android/telecom/InCallService to do what you want. We discussed with @manuquentin to eventually extend the scope of this lib, but we do nothing for the moment about that.

**Comment by danjenkins** [comment]
You can also call `backToForeground` when you know the call has been answered...... its not in the official readme but is available in the code.... not sure what ramifications are but I use it and it seems to do what I want in that scenario

**Comment by rph17** [comment]
@danjenkins when the app is in the background the backToForeground is working perfectly, now when the app is dead as a criterion to be done for it to call the app to the foreground?

**Comment by danjenkins** [comment]
You can't just bring it from dead to foreground unless you use the right notifications to do that (displaying a call using connection service is classed as that)

**Comment by rph17** [comment]
@danjenkins I am already listening to a notification on Android through firebase with the dead application behind a call call when receiving the desired direction to the application.

**Comment by rph17** [comment]
@danjenkins Error using backToForeground :
06-04 18:53:50.432  4509  4509 E unknown:ReactNative: Unable to launch redbox because react activity is not available, here is the error that redbox would've displayed: Attempt to invoke virtual method 'void android.app.Activity.startActivity(android.content.Intent)' on a null object reference

**Comment by danjenkins** [comment]
Right - you can't do that from an app that hasn't started up react native yet....  I'm unsure what you're trying to achieve here? How to make the app open from a push? It requires user interaction from a notification or your app to be given permission to draw over other apps

**Comment by rph17** [comment]
I need that when he answers the call in the background the app starts

**Comment by rph17** [comment]
Or do you want the application to be open in the background?

**Comment by maitzeth** [comment]
> You can also call `backToForeground` when you know the call has been answered...... its not in the official readme but is available in the code.... not sure what ramifications are but I use it and it seems to do what I want in that scenario

You should put this on the official documentation. I've lost like 5 hours for this method :(

**Comment by namnm** [comment]
@danjenkins As I understand, @rph17 was saying that if the phone is in background and not killed, then `backToForeground` will work. Otherwise if the phone is killed and receive a PN, `backToForeground` wont work.

So in short there no work around for this but only implement the `InCallService`?

**Comment by namnm** [comment]
@danjenkins Can you consider to reopen this issue, since @sboily mention that we could have this feature in the future?

**Comment by danjenkins** [comment]
There's another issue for implementing self managed connection service. #43 from what I remember

**Comment by danilvalov** [comment]
I'll just leave this comment here.

I had the same problem. And I found a solution here:
https://github.com/linhvovan29546/react-native-full-screen-notification-incoming-call
Example with react-native-callkeep integration:
https://github.com/linhvovan29546/react-native-full-screen-notification-incoming-call/blob/5d9ee46cb4f37e847ae0c237afd5c24108632f2a/example/src/services/CallKeepService.ts

Also my additional issues and solutions for them:
1. I had a problem when `RNCallKeep.displayIncomingCall` fired but `showIncomingCallUi` listener doesn't. I found a message in native android logs about calling `displayIncomingCall` with the following:
```
[RNCallKeepModule] sendEventToJS, eventName: RNCallKeepShowIncomingCallUi, bound: true, hasListeners: false
```
So I saw **hasListeners: false** and just moved `setupCallKeep()` to main `index.js` file of my react native app, and it fixed this issue.

2. My app needs to open the app by answer button pressing, so I tried to delete `READ_PHONE_NUMBERS` permission. So... we need this permission :)
